llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -2551,11 +2551,6 @@ void VPVectorPointerRecipe::printRecipe(raw_ostream &O, const Twine &Indent,
 
 InstructionCost VPBlendRecipe::computeCost(ElementCount VF,
                                            VPCostContext &Ctx) const {
-  // Handle cases where only the first lane is used the same way as the legacy
-  // cost model.
-  if (vputils::onlyFirstLaneUsed(this))
```

Comment:
could this more accurately use the scalar type below in this case isntead of the vector type?